### ISSUE TITLE
Firebase: Illegal App name when initializing an Ionic Angularfire2 app

### ISSUE THREAD

**Issue by giladrom** [angular/angularfire2#1831]
### Version info

**Angular:**
```
Angular CLI: 6.0.8
Node: 9.10.1
OS: darwin x64
Angular: 5.0.0
... common, compiler, compiler-cli, core, forms, http
... platform-browser, platform-browser-dynamic

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.6.8 (cli-only)
@angular-devkit/build-optimizer   0.0.35
@angular-devkit/core              0.6.8 (cli-only)
@angular-devkit/schematics        0.6.8 (cli-only)
@schematics/angular               0.6.8 (cli-only)
@schematics/update                0.6.8 (cli-only)
rxjs                              6.2.2
typescript                        2.4.2
webpack                           3.8.1
```

**Firebase:**
firebase@5.4.0

**AngularFire:**
angularfire2@5.0.0-rc.11

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**
Ionic 3.20.0
Cordova 8.0.0

### How to reproduce these conditions

**Failing test unit, Plunkr, or JSFiddle demonstrating the problem**

**Steps to set up and reproduce**
The following has worked using older Angularfire2/Firebase versions, but 
results in an error using the last version:

```
export const firebaseConfig = {
  apiKey: '...',
  authDomain: 'app.firebaseapp.com',
  databaseURL: 'https://app.firebaseio.com',
  projectId: 'app',
  storageBucket: 'app.appspot.com',
  messagingSenderId: '1234'
};

@NgModule({
  declarations: [MyApp],
  imports: [
    BrowserModule,
    IonicModule.forRoot(MyApp, {
      animate: true,
      preloadModules: true
    }),
    AngularFireModule.initializeApp(firebaseConfig)
...
})

```

**Sample data and security rules**

<-- include/attach/link to some json sample data (or provide credentials to a sanitized, test Firebase project) -->

### Debug output

** Errors in the JavaScript console **

```
[14:55:39]  console.error: Unhandled Promise rejection: Firebase: Illegal App name: '[object Object] (app/bad-app-name).
            ; Zone: <root> ; Task: Promise.then ; Value: [object Object]
            FirebaseError@http://192.168.0.170:8100/build/vendor.js:160363:36
            create@http://192.168.0.170:8100/build/vendor.js:160407:36
            error@http://192.168.0.170:8100/build/vendor.js:160001:27
            initializeApp@http://192.168.0.170:8100/build/vendor.js:159900:22
            http://192.168.0.170:8100/build/vendor.js:65648:104 run@http://192.168.0.170:8100/build/polyfills.js:3:10149
            AngularFirestore@http://192.168.0.170:8100/build/vendor.js:65647:48 AngularFirestore@[native code]
            _createClass@http://192.168.0.170:8100/build/vendor.js:10985:75
            _createProviderInstance$1@http://192.168.0.170:8100/build/vendor.js:10949:38
            initNgModule@http://192.168.0.170:8100/build/vendor.js:10902:53
            NgModuleRef_@http://192.168.0.170:8100/build/vendor.js:12145:21
            createNgModuleRef@http://192.168.0.170:8100/build/vendor.js:12135:28
            http://192.168.0.170:8100/build/vendor.js:5721:84 onInvoke@http://192.168.0.170:8100/build/vendor.js:4980:39
            run@http://192.168.0.170:8100/build/polyfills.js:3:10149
            run@http://192.168.0.170:8100/build/polyfills.js:3:10149
            http://192.168.0.170:8100/build/polyfills.js:3:20245
            runTask@http://192.168.0.170:8100/build/polyfills.js:3:10844
            o@http://192.168.0.170:8100/build/polyfills.js:3:7901 promiseReactionJob@[native code]
```

** Output from `firebase.database().enableLogging(true);` **

** Screenshots **

### Expected behavior

Angualrfire2 should initialize.

### Actual behavior
Angularfire2 does not initialize.

**Comment by giladrom** [comment]
For anyone experiencing the same issue - the problem was with the     
```
"@firebase/app": "^0.3.3",
```

dependency.

**Comment by kyleabens** [comment]
@giladrom So is the fix to update @firebase/app to 0.3.3?
